packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -0,0 +1,279 @@
+import React, {
+  useState,
+  useRef,
+} from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import { injectIntl, intlShape } from 'react-intl';
+import ThemeContext from 'terra-theme-context';
+import {
+  defaultPlaceholderValue,
+  isCurrentPlaceholder,
+  isValuePresent,
+  getDisplay,
+  getFirstValue,
+  getOptGroupKey,
+  getOptGroupKeyIndex,
+} from './NativeUtils';
+import {
+  optionPropType,
+  optGroupPropType,
+} from './NativePropTypes';
+
+import styles from './NativeSelect.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * Whether the placeholder is included as an option to clear selection.
+   * Dependent on the `'hasPlaceholder'` property.
+   */
+  allowClear: PropTypes.bool,
+  /**
+   * The string containing ids for elements to describe the select.
+   */
+  ariaDescribedBy: PropTypes.string,
+  /**
+   * The aria label string value for the select.
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * Additional attributes to spread directly onto the html select node.
+   * This prop is provided as standard customProps are spread onto the outer containing element which is not the html select in this case.
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  attrs: PropTypes.object,
+  /**
+   * The default value of the select. Can be a string, or number.
+   */
+  defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  /**
+   * Whether the input is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Whether a placeholder value should be included as an option.
+   */
+  hasPlaceholder: PropTypes.bool,
```

Comment:
@dkasper-was-taken I would rather this not be an exposed prop, since the placeholder text will always be defined internal to the component, only `allowClear` should have the ability to include the placeholder in the list of options and therefore this would be an internal only item to track with allowClear.